spring-cloud-gateway-webflux/src/main/java/org/springframework/cloud/gateway/webflux/ProxyExchange.java

```diff
@@ -376,7 +375,8 @@ private void addHeaders(HttpHeaders headers, HttpHeaders toAdd) {
 	}
 
 	private Set<String> filterHeaderKeys(HttpHeaders headers) {
-		return headers.keySet().stream().filter(header -> !sensitive.contains(header.toLowerCase()))
+		final Set<String> sensitiveHeaders = Optional.ofNullable(this.sensitive).orElse(DEFAULT_SENSITIVE);
```

Comment:
Please don't use `Optional` instead of an if statement. You could use the ternary operator if preferred.